docs/DeveloperGuide.md

```diff
@@ -1,38 +1,314 @@
+  - Cons: Significant time may be taken as timetable will have to be created from scratch. The delay may be extended if user has a lot of tasks.
+
+- Alternative Implementation: Having a few templates, before choosing the suitable template and editing it if needed.
+  - Pros: Reduces computation time
+  - Cons: Increases memory usage
+
+### Storage Implementation
+#### Loading saved files
+
+Class diagram of Storage:
+
+![](images/StorageClass.png)
+
+(_Note: some methods and attributes of `TaskList`,`Ui` and `Parser` are omitted here_)
+
+The storage component
+- Can save the content of a `TaskList` to a file in JSON format
+- Can load a JSON file to restore a previously saved `TaskList`
+- Relies on the `Parser` class to understand the content of a JSON file
+- Communicates with the user through the `Ui` class
+
+The loading of a save file is done with the function
+
+`Storage#load()` - Loads a saved JSON file and returns an ArrayList of task
+
+The path of the JSON file is provided as a parameter in the constructor of `Storage` hence 
+there is no need for any parameters in the `Storage#load()`. Since a save file will be created in the 
+constructor of `Storage` if no such file exists, there should not be any issue with a missing save file.
+
+The save file has the following fields:
+- `identifier`: A randomly generated number given to a task. Recurring tasks share the same identifier.
+- `index`: The index of the task in the array list
+- `description`: Description of the task
+- `status`: If the task is completed or not (`'X'` indicates completion, empty string otherwise)
+- `do_date_start`: The start date and time of the task (d/M/yyyy HH:mm format)
+- `do_date_end`: The end date and time of the task (d/M/yyyy HH:mm format)
+- `by_date`: The due date of the task (d/M/yyyy format)
+- `frequency`: How often the task repeats (`DAILY`,`WEEKLY`,`MONTHLY`,`NULL`)
+
+The sequence diagram of `Storage#load()` is shown here:
+![](images/StorageLoadSD.png)
+
+In the event where the save file cannot be parsed by `JSONObject` (i.e. the format of the file is incorrect) 
+or if there are missing fields in a task, the function `Storage#handleCorruptedSave()`
+will be called. The user will get to choose to create a new save file or exit the program for manual inspection.
+The file error will also be displayed to the user.
+
+The sequence diagram of `Storage#handleCorruptedSave()` is shown here:
 
+![](images/StorageCorruptedSD.png)
 
+#### Design considerations for the format of the save file
+- JSON (current choice)
+  - Pros: Easy to modify by hand if the user wants to
+  - Pros: Easy to parse data
+- XML
+  - Pros:Reasonably easy for users to modify
+  - Cons: Difficult to parse
+- Text file with custom format
+  - Pros: Might use less space (negligible)
+  - Cons: Unintuitive for users to modify
 ## Product scope
 ### Target user profile
```

Comment:
Take note that there is a minor formatting error here:
![image](https://user-images.githubusercontent.com/69451153/160969414-d0697387-127c-48f8-a5df-40542bc90941.png)